addons/mail/static/src/core/common/thread_service.js

```diff
@@ -1008,7 +1007,8 @@ export class ThreadService {
     async search(searchTerm, thread, before = false) {
         const { messages, count } = await this.rpc(this.getFetchRoute(thread), {
             ...this.getFetchParams(thread),
-            search_term: escape(searchTerm),
+            // treating the search string exactly like a string for a message to be posted
+            search_term: await prettifyMessageContent(searchTerm),
```

Comment:
Can put comment inline, so that this doesn't add a line of code.
Could be very brief. Suggestion:
```js
search_term: await prettifyMessageContent(searchTerm), // formatted like message_post
```